packages/inpage/setup-inpage-provider.js

```diff
@@ -12,31 +12,78 @@ function validateMessage(e) {
   if (typeof e.data.msg !== 'object') return
   if (e.data.msg.event) return true
   if (!e.data.msg.id && e.data.msg.id !== 0) return
-  if (!e.data.msg.error && !e.data.msg.result) return
+  if (!e.data.msg.error && e.data.msg.result === undefined) return
   if (e.data.msg.error && (!e.data.msg.error.message || !e.data.msg.error.code))
     return
 
   return true
 }
 
+const FLUENT_BACKEND_PREFERENCES = '__FLUENT_BACKEND_PREFERENCES__'
 const FLUENT_USE_MODERN_PROVIDER_API = '__FLUENT_USE_MODERN_PROVIDER_API__'
+const FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX =
+  '__FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX__'
+const FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM =
+  '__FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM__'
+
+function maybeUseModernProviderAPI(useModernProviderAPI) {
+  if (useModernProviderAPI) {
+    if (window.localStorage.getItem(FLUENT_USE_MODERN_PROVIDER_API)) return
+    window.localStorage.setItem(FLUENT_USE_MODERN_PROVIDER_API, true)
+    window.location.reload()
+  } else {
+    if (!window.localStorage.getItem(FLUENT_USE_MODERN_PROVIDER_API)) return
+    window.localStorage.removeItem(FLUENT_USE_MODERN_PROVIDER_API)
+    window.location.reload()
+  }
+}
+
+function maybeOverrideWindowDotConflux(overrideWindowDotConflux) {
+  if (overrideWindowDotConflux) {
+    if (
+      window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX) === false
+    ) {
+      window.localStorage.removeItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX)
+      window.location.reload()
+    }
+  } else {
+    if (
+      window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX) === false
+    )
+      return
+    window.localStorage.setItem(FLUENT_OVERRIDE_WINDOW_DOT_CONFLUX, false)
+    window.location.reload()
+  }
+}
+
+function maybeOverrideWindowDotEthereum(overrideWindowDotEthereum) {
+  if (overrideWindowDotEthereum) {
+    if (window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM)) {
```

Comment:
when window.localStorage.getItem(FLUENT_OVERRIDE_WINDOW_DOT_ETHEREUM), should return

Reply:
fixed with new push